hello-app-iap/main.go

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START all]
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	// use port1 serves service1 and port2 serves service2
+	const (
+		port1 = "8081"
+        	port2 = "8082"
+	)
+
+	finish := make(chan bool)
+
+	server1 := http.NewServeMux()
+	server1.HandleFunc("/", hello1)
+
+    	server2 := http.NewServeMux()
+	server2.HandleFunc("/", hello2)
+
+   	go func() {
```

Comment:
just do `go log.Fatal(http.ListenAndServe(...))` for simplicity here

Reply:
I believe http.ListenAndServe() should live in a literal, or it won't work, at least not in my test bed.

Reply:
Doesn't sound right.

You can do

```
go log.fatal(listenandserve(...))
go log.fatal(listenandserve(...))
<-ch //block forever
```

Reply:
Also ,you don't have to fork both servers in the background, you can do one foreground one background:

```
go log.Fatalf("failed to listen on server1: %+v", srv1.ListenAndServe())
log.Fatalf("failed to listen on server2: %+v", srv2.ListenAndServe())
```

Reply:
Hi, 
This is what I have done in the latest commit. Looks good?
```
go func() {
  log.Fatal(http.ListenAndServe(":"+port1, server1))
}()
log.Fatal(http.ListenAndServe(":"+port2, server2))
```